### PR TITLE
Add a position to the onTap callback

### LINKED ISSUE #1807: Unable to get touch locations in Tap, Pan or Scale events.
Use cases:
Tap on a map to place a pin.
Drag finger across the screen and leave a trail.
Show touch indicators on scale or pinch gestures (requires each finger's location).

The GestureScaleUpdateCallback has the focalPoint which is similar, but it would be nice to have access to each finger location in multi-finger versions of all of these.

### PR DESCRIPTION
**Pull request by abarth**
Fixes #1807

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie @krisgiesing

File: sky/packages/sky/lib/src/gestures/tap.dart [hunk 1]
```diff
@@ -11,7 +11,9 @@ import 'events.dart';
 import 'pointer_router.dart';
 import 'recognizer.dart';
 
-typedef void GestureTapCallback();
+typedef void GestureTapCallback(ui.Point globalPosition);
+typedef void GestureTapDownCallback();
+typedef void GestureTapCancelCallback();
```
Comment (krisgiesing) [inline sky/packages/sky/lib/src/gestures/tap.dart]: Why not just pass an event in each of these callbacks? Then we wouldn't have to add parameters in the future if someone cared about, say, the event timestamp.

Comment (Hixie) [inline sky/packages/sky/lib/src/gestures/tap.dart]: I don't think we should pass the full event data in the tap callback. The whole point of the gesture recognisers is to abstract out the raw data and present a fully formed gesture. If you want the raw data, you should be at the level below this, either directly listening to the pointer events (as InkWell does), or as a GestureRecogniser, translating the raw data into gestures.

**Comment by Hixie** [comment]
LGTM if you fix the skyanalyzer regexp to catch the bogus hint you're getting here.

Looking at the patch, though, I'm not sure this is the right way to go. We're making the common case (not caring about the coordinate) slightly more complicated, in favour of the rare case of wanting the coordinate. Why is it not better to have a separate gesture recogniser for the people who want the coordinates?

**Comment by Hixie** [comment]
LGTM

**Pull request merged**